src/Symfony/Component/DependencyInjection/Compiler/AutowirePass.php

```diff
@@ -115,50 +118,34 @@ private function completeDefinition($id, Definition $definition, $autowired)
     /**
      * Gets the list of methods to autowire.
      *
-     * @param \ReflectionClass     $reflectionClass
-     * @param true|string|string[] $autowired
+     * @param string           $id
+     * @param \ReflectionClass $reflectionClass
+     * @param string[]         $autowired
      *
      * @return \ReflectionMethod[]
      */
-    private function getMethodsToAutowire(\ReflectionClass $reflectionClass, $autowired)
+    private function getMethodsToAutowire($id, \ReflectionClass $reflectionClass, array $autowired)
     {
-        if (true === $autowired && $constructor = $reflectionClass->getConstructor()) {
-            return array($reflectionClass->getConstructor());
-        }
+        $found = array();
```

Comment:
It is, otherwise the array may be uninitialized (if the class has no method for instance).